rest_framework_json_api/utils.py

```diff
@@ -186,6 +186,14 @@ def get_related_resource_type(relation):
         relation_model = relation.model
     elif hasattr(relation, 'get_queryset') and relation.get_queryset() is not None:
         relation_model = relation.get_queryset().model
+    elif (
+            getattr(relation, 'many', False) and
+            hasattr(relation.child, 'Meta') and
```

Comment:
I'm wondering if these `hasattr` checks are a bit too much. A serializer doesn't have to be a `ModelSerializer` so there is no requirement for `Meta` to exist, but isn't `model` a required attribute of `Meta` for a `ModelSerializer`? It feels slightly too defensive to me to check that `model` exists.